controller/engine_controller.go

```diff
@@ -1777,6 +1777,27 @@ func (ec *EngineController) startRebuilding(e *longhorn.Engine, replicaName, add
 			log.Debug("Finished snapshot purge, will start rebuilding then")
 		}
 
+		// It has been at least 10 seconds (2 * EnginePollInterval) since we were asked to rebuild.
+		// Should we still communicate with addr?
+		updatedEngine, err := ec.ds.GetEngineRO(e.Name)
```

Comment:
I am not feeling strongly about getting a new version of engine CR in the middle of the sync loop of the old version of the engine CR and using the spec of the new version to make decision.

Maybe we should solve this 10s issue by implementing identity check for all the calls to engine/replica processes as outlined in [this ticket ](https://github.com/longhorn/longhorn/issues/5845)? What do you think?

Reply:
In my opinion it is worthwhile to do, though I understand that it breaks with purity of the design of the reconciliation loop. While https://github.com/longhorn/longhorn/issues/5845 should make this behavior obsolete, I am not sure it can be backported to stable versions of Longhorn. This "hacky" approach was sufficient by itself to prevent recreates of the inappropriate expansion every time I tried.

That being said, I'll defer to you and/or other reviewers on this one, @PhanLe1010 . If you'd still prefer to see it removed, I'll oblige.

Reply:
Thank you for the explanation! Lets keep this and backport it to older version and remove it once https://github.com/longhorn/longhorn/issues/5845 is implemented.